packages/api/amplify_api/test/ws/web_socket_connection_test.dart

```diff
@@ -36,78 +46,165 @@ void main() {
   }''';
   final subscriptionRequest = GraphQLRequest<String>(document: graphQLDocument);
 
-  setUp(() {
+  final startAck = WebSocketMessage(
+    messageType: MessageType.startAck,
+    id: subscriptionRequest.id,
+  );
+
+  final mockSubscriptionData = {
+    'onCreateBlog': {
+      'id': '<blog id>',
+      'name': 'Unit Test Blog',
+      'createdAt': '2022-09-12T18:03:36.230Z'
+    }
+  };
+
+  final mockSubscriptionEvent = {
+    'id': subscriptionRequest.id,
+    'type': 'data',
+    'payload': {
+      'data': mockSubscriptionData,
+    },
+  };
+
+  WebSocketChannel getWebSocketChannel(Uri uri) {
+    return MockWebSocketChannel();
+  }
+
+  MockWebSocketConnection getWebSocketConnection() {
+    MockClient mockClient = MockClient((request) async {
+      return http.Response('healthy', 200);
+    });
+
+    fakePlatform = MockConnectivityPlatform();
+    ConnectivityPlatform.instance = fakePlatform;
+    connectivity = Connectivity();
+
+    GraphQLSubscriptionOptions subscriptionOptions =
+        const GraphQLSubscriptionOptions();
+
     connection = MockWebSocketConnection(
       testApiKeyConfig,
       getTestAuthProviderRepo(),
+      subscriptionOptions: subscriptionOptions,
+      connectivityOverride: connectivity,
+      clientOverride: mockClient,
+      webSocketFactoryOverride: getWebSocketChannel,
+      logger: AmplifyLogger(),
     );
-  });
+
+    return connection;
+  }
+
+  Future<void> assertConnected() async {
+    await expectLater(connection.connectionPending, completes);
+
+    connection.channel!.sink.add(jsonEncode(mockAckMessage));
+
+    await expectLater(connection.ready, completes);
+
+    connection.channel!.sink.add(jsonEncode(startAck));
+  }
 
   group('WebSocketConnection', () {
-    test(
-        'init() should connect with authorized query params in URI and send connection init message',
-        () async {
-      await connection.init();
-      await expectLater(connection.ready, completes);
-      expect(
-        connection.connectedUri.toString(),
-        expectedApiKeyWebSocketConnectionUrl,
-      );
-      expect(
-        connection.lastSentMessage?.messageType,
-        MessageType.connectionInit,
-      );
+    setUp(() {
+      // hubEventsController = StreamController();
```

Comment:
mentioned offline but saw some weird hub event order. Am I correct in understanding still trying to fix that, then will uncomment these tests?